### PR TITLE
Fix macOS build when file provider module build is disabled

### PR DESCRIPTION
**Pull request by claucambra**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -190,6 +190,7 @@ endif()
 if (APPLE)
     # build macOS File Provider module
     option(BUILD_FILE_PROVIDER_MODULE "BUILD_FILE_PROVIDER_MODULE" ON)
+    add_definitions(-DBUILD_FILE_PROVIDER_MODULE=${BUILD_FILE_PROVIDER_MODULE})
```
Comment (mgallien) [inline CMakeLists.txt:193]: please put the definition in `config.h.in`

**Review by mgallien — CHANGES_REQUESTED** [review]
see comment

File: CMakeLists.txt (hunk 1 above)
Comment (claucambra) [reply CMakeLists.txt:193]: Done

Comment (mgallien) [reply CMakeLists.txt:193]: but if you put it in `config.h.in` then the explicit define is no longer needed or do I miss something ?

**Review by mgallien — CHANGES_REQUESTED** [review]

**Comment by claucambra** [comment]
I removed the CMakeLists add_definition call

**Review by mgallien — APPROVED** [review]

**Pull request merged**